gnovm/pkg/gnopkgfetch/gnopkgfetch.go

```diff
@@ -0,0 +1,130 @@
+package gnopkgfetch
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/gnolang/gno/gnovm/pkg/gnoimports"
+	"github.com/gnolang/gno/gnovm/pkg/gnolang"
+	"github.com/gnolang/gno/gnovm/pkg/gnomod"
+	tm2client "github.com/gnolang/gno/tm2/pkg/bft/rpc/client"
+	"github.com/gnolang/gno/tm2/pkg/commands"
+	"github.com/gnolang/gno/tm2/pkg/errors"
+	"golang.org/x/mod/module"
+)
+
+// FetchPackageImportsRecursively recursively fetches the imports of a local package while following a given gno.mod replace directives
+func FetchPackageImportsRecursively(io commands.IO, pkgDir string, gnoMod *gnomod.File) error {
+	imports, err := gnoimports.PackageImports(pkgDir)
+	if err != nil {
+		return fmt.Errorf("read imports at %q: %w", pkgDir, err)
+	}
+
+	for _, pkgPath := range imports {
+		resolved := gnoMod.Resolve(module.Version{Path: pkgPath})
+		resolvedPkgPath := resolved.Path
+
+		if !gnolang.IsRemotePkgPath(resolvedPkgPath) {
+			continue
+		}
+
+		depDir := gnomod.PackageDir("", module.Version{Path: resolvedPkgPath})
+
+		if err := fetchPackage(io, resolvedPkgPath, depDir); err != nil {
+			return fmt.Errorf("fetch import %q of %q: %w", resolvedPkgPath, pkgDir, err)
+		}
+
+		if err := FetchPackageImportsRecursively(io, depDir, gnoMod); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// fetchPackage downloads a remote gno package by pkg path and store it at dst
+func fetchPackage(io commands.IO, pkgPath string, dst string) error {
```

Comment:
Consider having this function fetch, and handle the save at the caller

Reply:
I feel this adds extra complexity for no benefits right now, pls see [this comment](https://github.com/gnolang/gno/pull/3123#discussion_r1854918503)